src/emit/substrate/target.rs

```diff
@@ -1183,6 +1209,12 @@ impl<'a> TargetRuntime<'a> for SubstrateTarget {
 
             binary.builder.position_at_end(bail_block);
 
+            self.log_runtime_error(
+                binary,
+                "contract creation failed".to_string(),
```

Comment:
we should probably include the error code. What do you think @xermicus

Reply:
Isn't the error code logged by ``log_api_return_values`` ?

Reply:
Ideally this would include the error code, but doesn't have to be perfect for me. @salaheldinsoliman is also right, while not super convenient, it is already available.